wormchain/x/wormhole/keeper/msg_server_allowlist_test.go

```diff
@@ -0,0 +1,249 @@
+	k.SetConfig(ctx, types.Config{
+		GovernanceEmitter:     vaa.GovernanceEmitter[:],
+		GovernanceChain:       uint32(vaa.GovernanceChain),
+		ChainId:               uint32(vaa.ChainIDWormchain),
+		GuardianSetExpiration: 86400,
+	})
+
+	createNewGuardianSet(k, ctx, guardians)
+	k.SetConsensusGuardianSetIndex(ctx, types.ConsensusGuardianSetIndex{
+		Index: 0,
+	})
+
+	context := sdk.WrapSDKContext(ctx)
+	msgServer := keeper.NewMsgServerImpl(*k)
+
+	// Test creating allowlist works using a validator
+	new_address := getRandomAddress()
+	_, err := msgServer.CreateAllowlist(context, &types.MsgCreateAllowlistRequest{
+		Signer:  getSigner(&guardians[0]),
+		Address: new_address,
+	})
+	assert.NoError(t, err)
+	// Test creating the same address again is rejected
+	for _, g := range guardians {
+		_, err = msgServer.CreateAllowlist(context, &types.MsgCreateAllowlistRequest{
+			Signer:  getSigner(&g),
+			Address: new_address,
+		})
+		assert.Error(t, err)
+	}
+
+	// Test address can be Deleted
+	_, err = msgServer.DeleteAllowlist(context, &types.MsgDeleteAllowlistRequest{
+		Signer:  getSigner(&guardians[0]),
+		Address: new_address,
+	})
+	assert.NoError(t, err)
+	// Can't be deleted again since it doesn't exist
+	_, err = msgServer.DeleteAllowlist(context, &types.MsgDeleteAllowlistRequest{
+		Signer:  getSigner(&guardians[0]),
+		Address: new_address,
+	})
+	assert.Error(t, err)
+	// Can be added again
+	_, err = msgServer.CreateAllowlist(context, &types.MsgCreateAllowlistRequest{
+		Signer:  getSigner(&guardians[0]),
+		Address: new_address,
+	})
+	assert.NoError(t, err)
+
+	// another guardian cannot delete an allowlist they did not create
+	for _, g := range guardians[1:] {
+		_, err = msgServer.DeleteAllowlist(context, &types.MsgDeleteAllowlistRequest{
+			Signer:  getSigner(&g),
+			Address: new_address,
+		})
+		assert.Error(t, err)
+	}
+
+	// Cannot make allowlist if not a validator
+	_, err = msgServer.CreateAllowlist(context, &types.MsgCreateAllowlistRequest{
+		Signer:  getRandomAddress(),
+		Address: getRandomAddress(),
+	})
+	assert.Error(t, err)
+
+	// Cannot make allowlist if the guardian set changes
+	oldGuardian := guardians[0]
+	guardians, _ = createNGuardianValidator(k, ctx, 10)
+	createNewGuardianSet(k, ctx, guardians)
+	err = k.TrySwitchToNewConsensusGuardianSet(ctx)
+	assert.NoError(t, err)
+	_, err = msgServer.CreateAllowlist(context, &types.MsgCreateAllowlistRequest{
+		Signer:  getSigner(&oldGuardian),
+		Address: getRandomAddress(),
+	})
+	assert.Error(t, err)
+
+	// still works with new guardian set
+	_, err = msgServer.CreateAllowlist(context, &types.MsgCreateAllowlistRequest{
+		Signer:  getSigner(&guardians[0]),
+		Address: getRandomAddress(),
+	})
+	assert.NoError(t, err)
+
+	// Anyone can remove stale allowlists
+	// (new_address list is now stale as it's validator is no longer in validator set)
+	_, err = msgServer.DeleteAllowlist(context, &types.MsgDeleteAllowlistRequest{
```

Comment:
One more test case here that would make sense is: New validator can "overwrite" existing stale entry according to handler#L21